tests/round_combinations/critical_path_analyzer.py

```
@@ -0,0 +1,290 @@
+        real_length_distribution = defaultdict(int)
+
+        # Track extreme paths without storing all
+        shortest_paths = []
+        longest_paths = []
+        max_appeal_paths = []
+        max_appeals_seen = 0
+
+        # File handle for saving paths if requested
+        path_file = None
+        if save_paths_to_file:
+            path_file = open(save_paths_to_file, "w")
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)